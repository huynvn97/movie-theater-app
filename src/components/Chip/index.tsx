import React from 'react';
import {
  Image,
  Pressable,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  ViewStyle,
} from 'react-native';
import {getImageUrl} from '@utils/image';

type ChipProps = {
  label: string;
  onPress?: () => void;
  containerStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<TextStyle>;
  leftIconUrl?: string;
};

export default function Chip({
  label,
  onPress,
  containerStyle,
  labelStyle,
  leftIconUrl,
}: ChipProps) {
  return (
    <Pressable onPress={onPress} style={[styles.container, containerStyle]}>
      {leftIconUrl && (
        <Image source={{uri: getImageUrl(leftIconUrl)}} style={[styles.icon]} />
      )}
      <Text style={[styles.label, labelStyle]}>{label}</Text>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#e0e0e0',
    paddingVertical: 6,
    paddingHorizontal: 12,
    borderRadius: 16,
    fontSize: 14,
    color: '#444',
    marginRight: 8,

    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {},
  icon: {
    width: 30,
    height: 30,
    borderRadius: 15,
    marginRight: 8,
  },
});
