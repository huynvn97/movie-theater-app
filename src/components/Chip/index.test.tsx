import React from 'react';
import {
  render,
  screen,
  userEvent,
  waitFor,
} from '@testing-library/react-native';
import Chip from './index'; // Adjust the import path as necessary

describe('Chip Component testing', () => {
  it('renders correctly - simple', () => {
    render(<Chip label="Test Chip" />);
    expect(screen).toMatchSnapshot();
  });

  it('renders correctly - has pressable', async () => {
    const fn = jest.fn();
    const user = userEvent.setup();

    render(<Chip label="Test Chip Press" onPress={fn} />);

    await user.press(screen.getByTestId('ChipPressableTestID'));

    expect(fn).toHaveBeenCalledTimes(1);
    expect(screen).toMatchSnapshot();
  });
});
