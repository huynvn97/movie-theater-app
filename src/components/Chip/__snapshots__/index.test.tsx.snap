// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Component testing renders correctly - has pressable 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#e0e0e0",
        "borderRadius": 16,
        "color": "#444",
        "flexDirection": "row",
        "fontSize": 14,
        "marginRight": 8,
        "paddingHorizontal": 12,
        "paddingVertical": 6,
      },
      undefined,
    ]
  }
  testID="ChipPressableTestID"
>
  <Text
    style={
      [
        {},
        undefined,
      ]
    }
  >
    Test Chip Press
  </Text>
</View>
`;

exports[`Chip Component testing renders correctly - simple 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#e0e0e0",
        "borderRadius": 16,
        "color": "#444",
        "flexDirection": "row",
        "fontSize": 14,
        "marginRight": 8,
        "paddingHorizontal": 12,
        "paddingVertical": 6,
      },
      undefined,
    ]
  }
  testID="ChipPressableTestID"
>
  <Text
    style={
      [
        {},
        undefined,
      ]
    }
  >
    Test Chip
  </Text>
</View>
`;
